exe-common/src/network/hermes.rs

```diff
@@ -26,29 +26,45 @@ impl HermesEndPoint {
     }
 
     pub fn uri(& mut self, path: &str) -> String {
-        format!("{}/{}/{}", self.url, self.blockchain, path)
+        format!("{}/{}", self.url, path)
     }
 }
 
 impl Api for HermesEndPoint {
     fn get_tip(&mut self) -> Result<BlockHeader> {
-        let uri = self.uri("tip").as_str().parse().unwrap();
+        let uri = self.uri("tip");
         info!("querying uri: {}", uri);
 
+        let mut err = None;
+
+        let now = SystemTime::now();
+
         let mut bh_bytes = Vec::with_capacity(4096);
         {
             let client = Client::new(&self.core.handle());
-            let work = client.get(uri).and_then(|res| {
-                res.body().for_each(|chunk| {
+            let work = client.get(uri.parse().unwrap()).from_err::<Error>()
+                .and_then(|res| {
+                println!("Response: {}", res.status());
+                if !res.status().is_success() {
+                    err = Some(Error::HttpError(uri, res.status().clone()));
+                };
+                res.body().from_err::<Error>().for_each(|chunk| {
                     bh_bytes.write_all(&chunk).map_err(From::from)
                 })
             });
-            let now = SystemTime::now();
             self.core.run(work)?;
-            let time_elapsed = now.elapsed().unwrap();
-            info!("Downloaded TIP in {}sec", time_elapsed.as_secs());
         }
 
+        let time_elapsed = now.elapsed().unwrap();
+
+        match err {
```

Comment:
good idea to handle the error here, one more elegant way to only get the error would be to use:

```rust
if let Some(err) = err {
  return Err(err);
}
```